### PR TITLE
Modify code runner Dockerfile to support ARM architecture

### PR DESCRIPTION
**Pull request by benallan**
Most of the code runner dockerfile was platform agnostic, but the Julia
installation was using a binary for x86_64 specifically.

I first re-wrote the docker file to use `$(uname -m)` to check the
platform, but the quoting and escaping in the RUN commands were a big
mess. Luckily there's a python tool called Jill which handles installing
Julia from the command line:
https://github.com/abelsiqueira/jill

**This needs tested on an x86 machine!** Unfortunately I don't have ready access to one right now, but maybe the CI pipeline will test this for us?

### CODE REVIEW

**Review by ali-ramadhan — APPROVED** [review]
Hey @benallan! This is great, looks good to merge. I didn't know about jill 👀 

> This needs tested on an x86 machine! Unfortunately I don't have ready access to one right now, but maybe the CI pipeline will test this for us?

I think so! I couldn't find a list of supported architectures for GitHub Actions but pretty sure the default is x86.

**Comment by ali-ramadhan** [comment]
I usually don't merge other people's PRs in case they want to modify things before merging (or want to merge in a specific way or time) but I hope you don't mind if I merge this so it makes it into v3.2.0 which I'm hoping to tag later today.

**Comment by benallan** [comment]
No problem, thanks for merging it!

**Pull request merged**
